clusterloader2/pkg/flags/flags.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"flag"
+	"os"
+
+	"github.com/spf13/pflag"
+)
+
+func init() {
+	pflag.CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+}
+
+var flags []flagEnv
+
+// StringVar creates string flag with given parameters.
+func StringVar(s *string, flagName, defaultValue, description string) {
+	pflag.StringVar(s, flagName, defaultValue, description)
+}
+
+// IntVar creates int flag with given parameters.
+func IntVar(i *int, flagName string, defaultValue int, description string) {
+	pflag.IntVar(i, flagName, defaultValue, description)
+}
+
+// StringEnvVar creates string flag with given parameters.
+// If flag is not provided, it will try to get env variable.
+func StringEnvVar(s *string, flagName, envVariable, defaultValue, description string) {
+	stringFlag := &stringFlagEnv{
+		v:            s,
+		envVariable:  envVariable,
```

Comment:
Given that you are introducing Var() anyway, this is too complicated..
It should work as following:
- read env-var (if it doesn't exist, use default)
- create a struct implementing Var interface that keeps just a single value equal to the one from above
- calling Set() is modifying that value

Reply:
I'm not sure why env value should be used first? IMHO, explicit flag sounds more like final setting provided by the user.

Reply:
Implementation changed to match what @wojtek-t described above. Priority for parameters: flag > env var > default.